app_build_suite/build_steps/helm.py

```diff
@@ -276,6 +278,82 @@ def run(self, config: argparse.Namespace, _: Context) -> None:
             raise BuildError(self.name, "Linting failed")
 
 
+class HelmRequirementsUpdater(BuildStep):
+    """
+    Executes helm dependency update.
+    """
+
+    _helm_bin = "helm"
+    _min_helm_version = "3.2.0"
+    _max_helm_version = "4.0.0"
+
+    @property
+    def steps_provided(self) -> Set[StepType]:
+        return {STEP_BUILD}
+
+    # noinspection PyMethodMayBeStatic
+    def _should_run(self, config: argparse.Namespace) -> bool:
+        return config.replace_chart_version_with_git
+
+    # noinspection PyMethodMayBeStatic
+    def _chart_lock_exists(self, config: argparse.Namespace) -> bool:
+        return os.path.isfile(os.path.join(config.chart_dir, _chart_lock))
+
+    def pre_run(self, config: argparse.Namespace) -> None:
+        """
+        Checks if the required version of helm is installed and if the lock file is present.
+        :param config: the config object
+        :return: None
+        """
+        if not self._should_run(config):
+            logger.debug("No chart version override requested, skipping dependency update.")
+            return
+        if not self._chart_lock_exists(config):
+            logger.debug(f"No {_chart_lock} file exists, skipping dependency update.")
+            return
+        self._assert_binary_present_in_path(self._helm_bin)
+        run_res = subprocess.run([self._helm_bin, "version"], capture_output=True)  # nosec
+        version_line = str(run_res.stdout.splitlines()[0], "utf-8")
+        prefix = "version.BuildInfo"
+        if version_line.startswith(prefix):
+            version_line = version_line[len(prefix) :].strip("{}")
+        else:
+            raise ValidationError(self.name, f"Can't parse '{self._helm_bin}' version number.")
+        version_entries = version_line.split(",")[0]
+        version = version_entries.split(":")[1].strip('"')
+        self._assert_version_in_range(self._helm_bin, version, self._min_helm_version, self._max_helm_version)
+
+    def run(self, config: argparse.Namespace, context: Context) -> None:
+        """
+        Runs 'helm dependencies update' to update or generate a Chart.lock file.
+        :param config: the config object
+        :param context: the context object
+        :return: None
+        """
+        context[context_key_chart_lock_changes_made] = False
+        chart_lock_path = os.path.join(config.chart_dir, _chart_lock)
+        if not self._should_run(config):
+            logger.debug("No chart version override requested. Dependency update not required, ending step.")
+            return
+        if not self._chart_lock_exists(config):
```

Comment:
This is already validate in `pre_run`. The idea is to do all the validation possible in `pre_run` to keep `run` shorter and focused on actual build logic, not validation. `pre_run` always runs before `run`, so I think re-validating here doesn't bring any benefit.

Reply:
But is there a way to skip the `run` function from returning something from the `pre_run`? Otherwise I don't see any way around executing this check in both `run` and `pre_run`

Reply:
Uhm, I don't really get your comment... If validation in `pre_run` fails, `run` is not executed at all. Is that what you mean?

Reply:
My question is: How should I tell the program to check for the `Chart.lock` file in the `pre_run` and tell it to skip `run` if no `Chart.lock` file is present?

Reply:
Aaa, now I got it. No strong opinion here, but to avoid code duplication, I used this pattern:

```python
class Blah(BuildStep):
  def init():
    self._should_run = True
   ...

  def pre_run():
    if not validate():
       self._should_run = False
  ...

  def run():
    if not self._should_run:
      return
  ....
```